indexing-service/src/test/java/org/apache/druid/indexing/overlord/setup/EqualDistributionWithCategorySpecWorkerSelectStrategyTest.java

```diff
@@ -187,4 +313,65 @@ private ImmutableWorkerInfo selectWorker(WorkerCategorySpec workerCategorySpec)
 
     return worker;
   }
+
+  /**
+   * Test implementation of SeekableStreamIndexTask for testing supervisor ID affinity
+   */
+  private static class TestSeekableStreamIndexTask extends SeekableStreamIndexTask<String, String, ByteEntity>
```

Comment:
`SeekableStreamSupervisorStateTest` also has a similar test class. Please see if it is possible to use the same one here. We can move it out into a file of its own.